File: chainOfResponsability/php/src/SoporteSegundoNivel.php
<?php
namespace SoporteTecnico;

class SoporteSegundoNivel implements Soporte {
    private ?Soporte $proximoNivel = null;

    public function setProximoNivel(Soporte $siguiente): void {
        $this->proximoNivel = $siguiente;
    }

    public function manejarSolicitud(string $solicitud): ?string {
        if (strpos($solicitud, 'hardware') !== false) {
            return "Soporte Técnico: Diagnosticando problema de hardware";
        }

        if ($this->proximoNivel !== null) {
            return $this->proximoNivel->manejarSolicitud($solicitud);
        }

        return "Ningún nivel de soporte puede manejar la solicitud";
    }
}

File: chainOfResponsability/pyhton/soporte/soporte_segundo_nivel.py
from soporte.soporte import Soporte

class SoporteSegundoNivel(Soporte):
    def manejar_solicitud(self, solicitud):
        if "hardware" in solicitud:
            return "Soporte Técnico: Diagnosticando problema de hardware"

        if self.proximo_nivel:
            return self.proximo_nivel.manejar_solicitud(solicitud)

        return "Ningún nivel de soporte puede manejar la solicitud"